tests/tst_ncio_mpi.F90

```diff
@@ -0,0 +1,225 @@
+! This is a test for the NCEPLIBS-ncio package.
+! Author: Brian Curtis (brian.curtis@noaa.gov) May 2021
+program tst_ncio_mpi
+
+  use mpi
+  use netcdf
+  use module_ncio
+  implicit none
+
+  character(len=72) charatt, time_units
+  type(Dataset) :: dset, dsetin
+  type(Variable) :: var
+  integer, parameter :: HALF_YT = 64, HALF_XT = 128 !yt = 128; xt = 256
+  integer, parameter :: MAXDIM = 3
+  real(4), allocatable, dimension(:) :: values_1d
+  real(4), allocatable, dimension(:,:) :: values_2d
+  real(4), allocatable, dimension(:,:,:) :: values_3d
+  real(4), allocatable, dimension(:,:,:,:) :: values_4d
+  real(4), allocatable, dimension(:,:,:,:,:) :: values_5d
+  real(4), dimension(10,10) :: quantize1, quantize2
+  real(4) mval,r4val,qerr
+  integer ndim,nvar,ndims,ival,idate(6),icheck(6),ierr,n,nbits
+  integer :: my_rank, nprocs
+  integer :: mpi_err
+  integer :: errcode
+  integer :: start(MAXDIM), count(MAXDIM)
+  logical hasit
+
+  call mpi_init(mpi_err)
+  call MPI_Comm_rank(MPI_COMM_WORLD, my_rank, mpi_err)
+  call MPI_Comm_size(MPI_COMM_WORLD, nprocs, mpi_err)
+
+  if (my_rank .eq. 0) print *, '*** Testing NCEPLIBS-ncio with MPI.'
+
+  if (nprocs .ne. 4) then
+    print *, 'This test must be run using only 4 processors.'
+    stop 1
+  endif
+
+  if (my_rank .eq. 0) print *, '*** Testing funtion open_dataset with paropen=.true.'
+  dsetin = open_dataset('dynf000_par_template.nc.in', paropen=.true.)
+
+  ! if (my_rank .eq. 0) print *,'*** Test creation of new dataset from template...'
+  ! dset = create_dataset('dynf000_par.nc', dsetin, paropen=.true.)
+
+  ! if (my_rank .eq. 0) print *,'*** Test that number of variables,dimensions,attributes is read...'
+  ! if (dsetin%nvars .ne. 24) stop 4
+  !
+  ! if (dsetin%ndims .ne. 6) stop 5
+  !
+  ! if (dsetin%natts .ne. 8) stop 6
+
+
+  if (my_rank .eq. 0) print *,'*** Test read of variable data...'
+  ! Let's set the starting point of data reads for each pe
+
+
+  count = (/ 0, HALF_YT, HALF_XT /)
+  if (my_rank .eq. 0) then
+     start = (/ 1, 1, 1 /)
+  else if (my_rank .eq. 1) then
+     start = (/ 1, HALF_YT + 1, 1 /)
+  else if (my_rank .eq. 2) then
+     start = (/ 1, 1, HALF_XT + 1 /)
+  else if (my_rank .eq. 3) then
+     start = (/ 1, HALF_YT + 1, HALF_XT + 1 /)
+  endif
+
+
+  call read_vardata(dsetin, 'pressfc', values_3d, ncstart=start, nccount=count, errcode=errcode)
```

Comment:
Is the errcode returned the netCDF error code? ...
I see from this code https://github.com/NOAA-EMC/NCEPLIBS-ncio/blob/develop/src/read_vardata_code_2d.f90 that it is.

So it would be useful to add:

`   print *, trim(nf90_strerror(errcode))

`(See example here: https://www.unidata.ucar.edu/software/netcdf/docs-fortran/f90_datasets.html#f90-nf90_strerror)